src/manifolds/KendallsPreShapeSpace.jl

```diff
@@ -0,0 +1,149 @@
+
+function active_traits(f, ::KendallsPreShapeSpace, args...)
+    return merge_traits(IsEmbeddedSubmanifold())
+end
+
+representation_size(::KendallsPreShapeSpace{n,k}) where {n,k} = (n, k)
+
+"""
+    check_point(M::KendallsPreShapeSpace, p; atol=sqrt(max_eps(X, Y)), kwargs...)
+
+Check whether `p` is a valid point on [`KendallsPreShapeSpace`](@ref), i.e. whether
+each row has zero mean. Other conditions are checked via embedding in [`ArraySphere`](@ref).
+"""
+function check_point(M::KendallsPreShapeSpace, p; atol=sqrt(eps(eltype(p))), kwargs...)
+    for p_row in eachrow(p)
+        if !isapprox(mean(p_row), 0; atol, kwargs...)
+            return DomainError(
+                mean(p_row),
+                "The point $(p) does not lie on the $(M) since one of the rows does not have zero mean.",
+            )
+        end
+    end
+    return nothing
+end
+
+"""
+    check_vector(M::KendallsPreShapeSpace, p, X; kwargs... )
+
+Check whether `X` is a valid tangent vector on [`KendallsPreShapeSpace`](@ref), i.e. whether
+each row has zero mean. Other conditions are checked via embedding in [`ArraySphere`](@ref).
+"""
+function check_vector(M::KendallsPreShapeSpace, p, X; atol=sqrt(max_eps(X, Y)), kwargs...)
+    for X_row in eachrow(X)
+        if !isapprox(mean(X_row), 0; atol, kwargs...)
+            return DomainError(
+                mean(X_row),
+                "The vector $(X) is not a tangent vector to $(p) on $(M), since one of the rows does not have zero mean.",
+            )
+        end
+    end
+    return nothing
+end
+
+embed(::KendallsPreShapeSpace, p) = p
+embed(::KendallsPreShapeSpace, p, X) = X
+
+function get_embedding(::KendallsPreShapeSpace{N,K}) where {N,K}
+    return ArraySphere(N, K)
+end
+
+@doc raw"""
+    manifold_dimension(M::KendallsPreShapeSpace)
+
+Return the dimension of the [`KendallsPreShapeSpace`](@ref) manifold `M`. The dimension is
+given by ``n(k - 1) - 1``.
+"""
+manifold_dimension(::KendallsPreShapeSpace{n,k}) where {n,k} = n * (k - 1) - 1
+
+"""
+    project(M::KendallsPreShapeSpace, p)
+
+Project point `p` from the embedding to [`KendallsPreShapeSpace`](@ref) by selecting
+the right element from the orthogonal section representing the quotient manifold `M`.
+See Section 3.7 of [^Srivastava2016] for details.
+
+# References
+
+[^Srivastava2016]:
+    > A. Srivastava and E. P. Klassen, Functional and Shape Data Analysis. Springer New York, 2016.
+"""
+project(::KendallsPreShapeSpace, p)
+
+function project!(::KendallsPreShapeSpace, q, p)
+    q .= p .- mean(p, dims=2)
+    q ./= norm(q)
+    return q
+end
+
+"""
+    project(M::KendallsPreShapeSpace, p, X)
+
+Project tangent vector `X` at point `p` from the embedding to [`KendallsPreShapeSpace`](@ref)
+by selecting the right element from the tangent space to orthogonal section representing the
+quotient manifold `M`. See Section 3.7 of [^Srivastava2016] for details.
+
+# References
+
+[^Srivastava2016]:
```

Comment:
Oh and maybe one Footnote is enough? Currently this might render as 17-or-so footnotes if it is set in every docstring.

Reply:
I've reduced the number of footnotes but it's a bit unfortunate that then they don't appear in docstrings :(.

Reply:
well, the only other solution is know would be to switch to DocumenterCite and have one page of references, but then they also would not show up in the individual doc strings (actually in none of them). We could also put the footnote just into the main function where this is needed?

Reply:
Ideally this would get merged: https://github.com/JuliaDocs/Documenter.jl/pull/1464 and solve the root of the problem.

Reply:
Will check that one, but at first glance my problem would be: Which footnote wins?

Reply:
If they are identical, why would it matter?

Reply:
right I read the topic and sure if they are identical then one is enough.